### ISSUE TITLE
Error when trying to upload a nightly release

### ISSUE THREAD

**Issue by juliushaertl** [ChristophWurst/krankerl#20]
I just tried to release a nightly build of deck to the app store, but if fails with the following error message:

```
$  krankerl publish --nightly https://download.bitgrid.net/nextcloud/deck/nightly/deck_nightly_2017-12-07.tar.gz
an error occured: Other("No package found")
```

@ChristophWurst

**Comment by juliushaertl** [comment]
Ah I needed to package the nightly build with krankerl as well. I guess it should be possible to use a prebuild package url and just download it if it is not present in the build/artifacts/ directory.

**Comment by ChristophWurst** [comment]
Did you run `krankerl package` before? Does `build/artifacts/<appid>.tar.gz` exist?

The error originates from https://github.com/ChristophWurst/krankerl/blob/a5e345b2a9a11354b82a0f77a18601e62bb65f02/src/lib.rs#L84.

**Comment by ChristophWurst** [comment]
> Ah I needed to package the nightly build with krankerl as well. Maybe it should be possible to use a prebuild package url and just download it if it is not present in the build/artifacts/ directory.

Yup, right. What package URL do you mean?

**Comment by juliushaertl** [comment]
Well, I run a remote cron to build nightly tarballs so i just tried to upload it as a release with krankerl without building it locally.

**Comment by ChristophWurst** [comment]
Right now you could trick krankerl by simply downloading the tarball to `build/artifacts`. I could be wrong but it should be possible to release a nightly build this way, although it's a bit hacky.

**Comment by juliushaertl** [comment]
Yep, that worked fine. I guess it kind of makes sense from a security point of view to require a local build. Maybe it should not download by default but a possible download CLI parameter would be great, or at least showing an error message that gives a bit more detail on what's wrong.

**Comment by ChristophWurst** [comment]
> not download by default but a possible download CLI parameter would be great

Something like `krankerl publish --nightly --download <url>`? Then, if the `--download` flag is set, it does not use the local build, but download the file from the given URL, sign it and register the release on the app store.

**Comment by juliushaertl** [comment]
Exactly. That would give the possibility to do so but not download it automatically.

**Comment by ChristophWurst** [comment]
I'm considering making this the default. Because I noticed a little flaw in the current process: the tool uses the locally available artifact to generate the tarball signature.

This fails
* Package tarball 1
* Upload tarball 1 to github
* Package another tarball 2
* Publish with URL from github asset
* :boom:  app store can't verify signature
